app/components/swiper.js

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    InteractionManager,
+    View,
+    ScrollView,
+    ViewPagerAndroid,
+    Platform,
+    StyleSheet
+} from 'react-native';
+
+export default class Swiper extends PureComponent {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        style: PropTypes.object,
+        showsPagination: PropTypes.bool,
+        paginationStyle: PropTypes.object,
+        initialPage: PropTypes.number,
+        width: PropTypes.number,
+        dotColor: PropTypes.string,
+        activeDotColor: PropTypes.string,
+        keyboardShouldPersistTaps: PropTypes.string,
+        onIndexChanged: PropTypes.func
+    };
+
+    static defaultProps = {
+        removeClippedSubviews: true,
+        showsPagination: true,
+        initialPage: 0,
+        keyboardShouldPersistTaps: 'handled',
+        onIndexChanged: () => null
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.runOnLayout = true;
+        this.state = this.initialState(props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState(this.initialState(nextProps));
```

Comment:
Will this trigger a re-render if none of the props that go into the state change? It might be good to surround this with an if statement just in case